skopt/space/space.py

```diff
@@ -721,3 +724,108 @@ def distance(self, point_a, point_b):
             distance += dim.distance(a, b)
 
         return distance
+
+    def get_dimension(self, id):
```

Comment:
What about using a `dict` like interface so that `some_space['learning_rate']` returns the dimension called `learning_rate`. We might not need to support indexing based access as people can to `some_space.dimensions[idx]`. This way we avoid having to support `some_space[idx]` and `some_space['learning_rate']` which might require us making guesses about what the user meant.

Reply:
Nice idea I'll look into it!

Reply:
I read this again and you make a very good point.

Reply:
This works now using `Space.__getitem__()`.